### PR TITLE
goonchem fixes

### PR DESCRIPTION
**Pull request by Iamgoofball**
fixes cyanide for reals this time

also makes morphine work as a painkiller

and its added to poppies

ambrosia deus now dispenses omnizine

omnizine doesn't heal no more, it's needed for recipes tho

adds hunchback

makes calomel have the proper depletion rate

nerfs salglu for being too OP

### CODE REVIEW

File: code/modules/mob/living/carbon/shock.dm [hunk 1]
```diff
@@ -21,6 +21,8 @@
 		src.traumatic_shock -= 50
 	if(reagents.has_reagent("tramadol"))
 		src.traumatic_shock -= 80 // make synaptizine function as good painkiller
+	if(reagents.has_reagent("morphine"))
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/shock.dm]: So this is not OOP.

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/shock.dm]: welcome to how the chemistry code works

**Comment by nuklearcellphoneg** [comment]
I don't know where to put this, so I'll put it here. I only looked at the medicines, but here is my two bits on some balance issues brought on by this. I don't think this new code is being added with any consideration at all to the current system in place. It's all pretty easy to make, I just pumped out bunches of all of it that I tried to.

With the medicine, you can make cryo heal from 16+ damage of all types per tick with these new chems(some are damage and RNG dependent, but it still heals fast).

Same applied with placing it into pills, syringes, rags, patches, and so on. It takes things to TG levels of easy healing.

Further, some of these chems are quite redundant- not that they have no place or are ineffectual necessarily, but that they are added in seemingly with the sole purpose of adding in goon chems. Hyronalin and Arithrizine are essentially analogous to Potassium Iodide and Pentetic acid. Salbutamol is alikel to Dexalin and Perflurodecalin is an unnecessary upgrade that is dwarfed by dexalin plus. Morphine is the same as Codeine except it has addiction properties. It would be better, perhaps easier, to give codeine and tramadol new addiction properties rather than adding in a new chem. Mannitol is analogous to Alkysine, Mannitol is identical to Ethyldrizcedine or whatever it is that removes alcohol from your system.

In short, it's very silly to me to try and make these huge sweeping changes when first you should change the existing chems to match these changes(adding in addiction, yadda yadda). Further, there is no need whatsoever to add in new reagents for these new recipes; make new recipes out of the existing reagents. 

Furthermore, for gameplay purposes, it would be much much much better to, instead of making these things into stuff that needs to be created in chemistry, to make it into things that only botany can produce, for example, styptic powder would be a great thing for botany to be capable of making. Same with Silver Sulfiadizine. Oh, but what does Silver have to do with botany? Well, then don't just copy the nonsense straight from goon and shove it in here, turn it into a new thing that is derived from x plant. Make it where they can grow cotton, in order to make rags, in order to apply the new powder and such.

Fairly redundant goon chems with analogs to our own:

Potassium Iodide: Chance to remove rads. Hyronalin removes rads and heals toxin. Not too redundant. 80% chance.

Pentetic Acid: Removes 7 rads, heals for toxin by four. Chance to do 1 brute damage per tick. Arithrizine removes all radiation, heals some toxin damage, does .5 ish brute damage per tick.

Salbutamol: Heals 6 oxygen, reduces lose breath. Another form of dexalin.

Perflurodecalin: Heals 25 oxy damage but can't talk. Dexalin Plus heals all oxy damage and can be made from Dexalin.

Ephedrine: Reduces stun, chance for addiction. No addiciton present in the mechanics yet; kind of like hyperzine.

Diphenhydramine: Makes you drowsy, makes you not jittery(see Hyperzine effects) removes histamines(wut)

Morphine: Same as Codeine, stronger than Tramadol, except it had addiction properties.

Oculine: Identical to imidizoline except it has a chance not to work.

Life: Chemical reaction which creates mobs; identical to the thing you do with slime cores

Mannitol: Cures brain damage, identical to alkysine

Antihol: Identical to Ethylnidrizine, or whatever it is that removes alcohol.

**Comment by Iamgoofball** [comment]
A good amount of the chemicals are still in because they are required in other recipes, so that it is more separate from trekchem to emphasize that the two systems aren't replacing eachother. 

As I said in the /vg/ thread, histamine is a toxin available to traitors and a byproduct of some other chemicals, so histamine is a thing.

**Comment by ComicIronic** [comment]
So why not merge those similar chems together and have the oldchems replace the new ones in recipes?

**Comment by Iamgoofball** [comment]
@nuklearcellphoneg 
what's your mix of chemicals that got you 16p of each damage type healing per cycle in cryo

**Comment by AutismSpider** [comment]
Burn:
Kelotane
Dermaline
Tricord
DD
Ointment
Natural
Tomato juice (same teat % base)

Brute:
Bicaridine
Natural
Bandage
Ointment
Tricord
DD

Oxygen:
Dexalin
Dexalin Plus

Toxin:
Tea (this % base i think was 30%?)
Anti Toxin
Hyronalin
Arithrizine
Tricord
DD
Charcoal

This is the number of chems we have currently on live that heal the same thing and stack with each other.

Now with goofchem
Burn:
Silver Sulfiadizine(External only) 2
Saline-Glucose(33% chance) 3
SynthMeat 100% 1 point
Omnizine 1 point
Perflurodecalin 33% chance to heal 1 brute/burn
Brute:
Styptic Powder(external only) 2
Saline-Glucose(33% chance) 3
SynthMeat 100% 1 point
Omnizine 1 point (gone now)
Salicyclic Acid If brute damage < 50, 50% chance to heal 1 brute
Perflurodecalin 33% chance to heal 1 brute/burn

Oxygen:
Omnizine 1 point (gone now)
Salbutamol Heals 6 oxy damage, reduces lose breath
Perflurodecalin Heals 25 oxy damage but can't talk, 33% chance to heal 1 brute/burn

Toxin:
Charcoal 3 tox, purges chems, identical to activated charcoal
Omnizine 1 point (gone now)
Potassium Iodide: 80% to remove 1 rad (hyro is better)
Pentetic Acid: Removes 7 rads, heals 4 toxin, 33% of 1 brute damage

**Comment by Iamgoofball** [comment]
>perf

No, you're wrong about perf.

Perf is healing 25 OXY per tick but mutes you.

**Comment by AutismSpider** [comment]
if it works that way then its bit better then Dexalin

**Comment by nuklearcellphoneg** [comment]
Dexalin is readily available throughout the station. One dexalin pill + 30u iron + 30u carbon gives you 90u dexalin plus.

Or if you really want you can derive dexalin from scratch, and make dexalin plus from that. Dexalin plus is still much better and more easily obtained than dexalin plus.

**Comment by nuklearcellphoneg** [comment]
Kelotane 1
Dermaline 1.5
Tricord %
DD %
Ointment 1-2?
Natural Low, let's say .2, I think it's spread out or delayed
Tomato juice %
Saline-Glucose(33% chance) 3
SynthMeat 100% 1 point
Omnizine 1 point
Perflurodecalin 33% chance to heal 1 brute/burn

Depending on how much damage is accrued(since some are based on percentages) the damage healed varies. But doing a minimum of 1 for each, you have about 14 here. Plus cryox and clonex, since they're in cryo, for 16 per tick, though in reality it's like much more.

All I'm saying is that there is no need to add ALL of the chems from goon. There's no need to add ALL of them to chemistry. A lot of them could be added to botany to add a whole new level of fun and usefulness to them.

Further, again, it would be better to make some chems with new applications that are vg specific. For example, there's some chem heater thing added. I don't know if it can cool things as well, but if it can, say you have 1 unit of kelotane and dermaline in a beaker, add it to the heater thing. Super chill it(since it's a burn treatment) and it combines to make a new chem that also heals burn damage, but can be stacked together with kelotane and dermaline.

Stuff like that would be cool, because it adds to and expands upon the current system. This is just a straight up port with a bunch of conflicting chems and balancing issues.

**Comment by ComicIronic** [comment]
And Bob Dylan manages to exploit a system to full potential, :+1:.
Worried about the impact of improved cryostack on an already cryo-heavy medbay.

**Comment by nuklearcellphoneg** [comment]
Well, it's not cryo dependent. It can be done with large/space beakers and IVs, it can also be done with to a lesser extent with pills.

Just by looking at the medicines that are added, they're added with no real regard to our current system of medicines. I'm all for variety and all, but we have alkysine, do we need mannitol?

We have Ethylredoxrazine, do we need Antihol?

We have Codeine, do we need Morphine?

I like the idea of addiction. I like the idea of having a larger variety of chemicals to make. But just taking on an entire separate system of chemicals onto our own isn't the best way to do it. At the very least pick and choose.

Out of every single of the medicines I looked at, the only two I really liked were the styptic powder and silver sulfiwhatever. And honestly I think it would be better to make an aloe plant or something and make that something botany can make.  This is really and honestly just adding goon chem for the sake of adding goon chem, with no regard to our own system.

**Comment by AutismSpider** [comment]
Thing is that only work with cyro and will heal way more then he listed.

After stacking all the burn meds types on a patch it keep plasma man form burning to death.

What i like to see is less cyro tube and more pills(and patches if we get goofchem stuff) but this will never happen.

**Comment by Iamgoofball** [comment]
>Perflurodecalin 33% chance to heal 1 brute/burn

no this is wrong stop

**Comment by nuklearcellphoneg** [comment]
I'm going by what you wrote in the code.

**Comment by Duny-** [comment]
REMOVE GOOFBALL

**Comment by Iamgoofball** [comment]
man this is what happens when i don't look over my code too well

**Comment by AutismSpider** [comment]
Getting my number off the code if its wrong want look over all of it then.

**Pull request merged**